### ISSUE TITLE
[enhancement]: add vae precision to image metadata

### ISSUE THREAD

**Issue by StefanTobler** [invoke-ai/InvokeAI#5074]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Contact Details

@evernbro

### What should this feature add?

VAE Model precision should be added to the VAE object in a generated image's metadata. This would allow for the precision used to generate the image to be recalled such that images can more consistently be reproduced.

This could be an extension of the feature/bugfix in #5073

**Comment by ufuksarp** [comment]
Don't wanna open another issue for this, but shouldn't Highres fix settings, noise settings also be in the metadata?

**Comment by StefanTobler** [comment]
Agreed, can you please help me out and list all the additional fields you want to see in the metadata? I can update the issue to reflect an overall update to the metadata fields.

**Comment by ufuksarp** [comment]
> Agreed, can you please help me out and list all the additional fields you want to see in the metadata? I can update the issue to reflect an overall update to the metadata fields.

Currently nothing inside High Resolution Fix and Advanced tab is saved. Those should definitely be saved in entirety.

**High Resolution Fix:**
Enable High Resolution Fix, Initial Width, Initial Height, Denoising Strength (which has some sort of floating point problem btw)

**Advanced:**
Seamless Tiling X Axis, Y Axis, CLIP Skip level, CPU/GPU Noise
